ddtrace/contrib/flask/patch.py

```diff
@@ -560,14 +569,20 @@ def _traced_request(pin, wrapped, instance, args, kwargs):
         # This call may be unnecessary since we try to add the tags earlier
         # We just haven't been able to confirm this yet
         _set_request_tags(span)
+        request = flask.request
 
+        _asm_context.asm_request_context_set(request.remote_addr, request.headers)
```

Comment:
For Flask is a little different than Django since we need to clear the outside this [here](https://github.com/DataDog/dd-trace-py/pull/4613/files/7b8853882dac1ea7ef10b4db5582e7dea1e21586#diff-83b4eda0004838863b6f460c5fb4f05ba87dd6d16192db38108c1973e2937383R157); clearing the context here would reset it before getting the response tags. So we set the values here, when we have them, and reset outside this. Thus we can't use the context manager here. But maybe there is a better way? (I admit I'm less familiar with Flask than Django).